dotCMS/src/main/java/com/dotcms/storage/DataBaseStorage.java

```diff
@@ -0,0 +1,260 @@
+                ).getOrElse(false);
+    }
+
+    @CloseDBIfOpened
+    @Override
+    public List<Object> listGroups() {
+
+        final List<Map<String, Object>> results = Try.of(()->new DotConnect().setSQL("select storage_group from storage")
+                .loadObjectResults()).getOrElse(()-> Collections.emptyList());
+
+        return results.stream().map(map -> map.get("storage_group")).collect(CollectionsUtils.toImmutableList());
+    }
+
+    @WrapInTransaction
+    @Override
+    public Object pushFile(final String groupName, final String path,
+                           final File file, final Map<String, Object> extraMeta) {
+
+        try (final FileByteSplitter fileSplitter = new FileByteSplitter(file)) {
+
+            final HashBuilder objectHashBuilder = Encryptor.Hashing.sha256();
+            final List<String> chuckHashes      = new ArrayList<>();
+
+            for (final Tuple2<byte[], Integer> bytesRead : fileSplitter) {
+
+                objectHashBuilder.append(bytesRead._1(), bytesRead._2());
+                final String chuckHash = Encryptor.Hashing.sha256().append
+                        (bytesRead._1(), bytesRead._2()).buildUnixHash();
+                chuckHashes.add(chuckHash);
+                new DotConnect().executeUpdate("insert into storage_objects_data(hash_id, object) values (?, ?)",
+                        chuckHash,
+                                    bytesRead._1().length == bytesRead._2()?
+                                        bytesRead._1(): this.chuckBytes (bytesRead._2(), bytesRead._1()));
+            }
+
+            final String objectHash = objectHashBuilder.buildUnixHash();
+            new DotConnect().executeUpdate("insert into storage(hash_id, key, storage_group) values (?, ?, ?)",
+                    objectHash, path, groupName);
+
+            int order = 1;
+            for (final String chuckHash : chuckHashes) {
+
+                new DotConnect().executeUpdate("insert into storage_objects(storage_hash, data_hash, data_order) values (?, ?, ?)",
+                        objectHash, chuckHash, order++);
+            }
+
+            return true;
+        } catch (DotDataException | NoSuchAlgorithmException | IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private  byte[]  chuckBytes(final int bytesLength, final byte[] bytes) {
+
+        return bytes; // todo: create a bytes array with the bytesLength
+    }
+
+    private byte[] fileToBytes(final File file) {
+
+        try (InputStream input = com.liferay.util.FileUtil.createInputStream(file.toPath(), "none")) {
+
+            final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();  // todo not sure if I should compress this
+            final byte[] buffer = new byte[1024];
+            for (int readNum; (readNum = input.read(buffer)) != -1;) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)